src/osd/ECBackend.cc

```diff
@@ -2342,13 +2344,23 @@ int ECBackend::send_all_remaining_reads(
   GenContext<pair<RecoveryMessages *, read_result_t& > &> *c =
     rop.to_read.find(hoid)->second.cb;
 
+  // (Note cuixf) If we need to read attrs and we read failed, try to read again.
+  bool want_attrs = 
+    rop.to_read.find(hoid)->second.want_attrs && 
+    (!(rop.complete[hoid].attrs) ||
+    (rop.complete[hoid].attrs && rop.complete[hoid].attrs->empty()));    
+  if (want_attrs) {
+    dout(10) << __func__ << " want attrs again=" << want_attrs << " "
```

Comment:
`want_attrs` is alway true here, what's the point to print it?

Reply:
`want_attrs` is not always true, it is false at most time.  print it to tell us that something bad happened but we have fixed it. maybe we can just print 'want attrs again' to make it shorter?

Reply:
i mean it is always true here -- the line above.

> maybe we can just print 'want attrs again' to make it shorter?

yeah.